rs/bitcoin/checker/src/main.rs

```diff
@@ -134,11 +139,19 @@ fn init(arg: CheckArg) {
 fn post_upgrade(arg: CheckArg) {
     match arg {
         CheckArg::UpgradeArg(arg) => {
-            if let Some(check_mode) = arg.and_then(|arg| arg.check_mode) {
-                let config = Config::new_and_validate(get_config().btc_network(), check_mode)
+            let old_config = get_config();
+            let num_subnet_nodes = arg
+                .as_ref()
+                .and_then(|arg| arg.num_subnet_nodes)
+                .unwrap_or(old_config.num_subnet_nodes);
+            let check_mode = arg
+                .as_ref()
+                .and_then(|arg| arg.check_mode)
+                .unwrap_or(old_config.check_mode);
+            let config =
+                Config::new_and_validate(get_config().btc_network(), check_mode, num_subnet_nodes)
```

Comment:
nit: we can re-use `old_config`:

```suggestion
                Config::new_and_validate(old_config.btc_network(), check_mode, num_subnet_nodes)
```

Reply:
Suggestion applied